integration-cli/docker_api_logs_test.go

```diff
@@ -82,3 +86,96 @@ func (s *DockerSuite) TestLogsAPIContainerNotFound(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, http.StatusNotFound)
 }
+
+func (s *DockerSuite) TestLogsAPIUntilFutureFollow(c *check.C) {
```

Comment:
Okay, I got this working on both JSON and journald. For the latter I had to introduce a new return variable to `followLogs` in order to terminate the `for` loop. I wasn't sure whether doing this in the C `wait_for_data_cancelable` function was possible so I opted for this. LMKWYT